streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignorTest.java

```diff
@@ -0,0 +1,347 @@
+        assertThat(clients.get(p3).activeTasks(), equalTo(Collections.singleton(task01)));
+    }
+
+    @Test
+    public void shouldAssignTasksToClientWithPreviousStandbyTasks() throws Exception {
+        final ClientState<TaskId> client1 = createClientWithPreviousActiveTasks(p1, 1);
+        client1.addPreviousStandbyTasks(Utils.mkSet(task02));
+        final ClientState<TaskId> client2 = createClientWithPreviousActiveTasks(p2, 1);
+        client2.addPreviousStandbyTasks(Utils.mkSet(task01));
+        final ClientState<TaskId> client3 = createClientWithPreviousActiveTasks(p3, 1);
+        client3.addPreviousStandbyTasks(Utils.mkSet(task00));
+
+        final StickyTaskAssignor taskAssignor = createTaskAssignor(task00, task01, task02);
+
+        taskAssignor.assign(0);
+
+        assertThat(clients.get(p1).activeTasks(), equalTo(Collections.singleton(task02)));
+        assertThat(clients.get(p2).activeTasks(), equalTo(Collections.singleton(task01)));
+        assertThat(clients.get(p3).activeTasks(), equalTo(Collections.singleton(task00)));
+    }
+
+    @Test
+    public void shouldAssignBasedOnCapacityWhenMultipleClientHaveStandbyTasks() throws Exception {
+        final ClientState<TaskId> c1 = createClientWithPreviousActiveTasks(p1, 1, task00);
+        c1.addPreviousStandbyTasks(Utils.mkSet(task01));
+        final ClientState<TaskId> c2 = createClientWithPreviousActiveTasks(p2, 2, task02);
+        c2.addPreviousStandbyTasks(Utils.mkSet(task01));
+
+        final StickyTaskAssignor taskAssignor = createTaskAssignor(task00, task01, task02);
+
+        taskAssignor.assign(0);
+
+        assertThat(clients.get(p1).activeTasks(), equalTo(Collections.singleton(task00)));
+        assertThat(clients.get(p2).activeTasks(), equalTo(Utils.mkSet(task02, task01)));
+    }
+
+
+    @Test
+    public void shouldAssignToClientWithStandbyIfProcessWithPreviousReachedCapacity() throws Exception {
+        final ClientState<TaskId> client = createClientWithPreviousActiveTasks(p1, 1, task00, task01);
+        // give p1 assignment so already at capacity
+        client.assign(task02, true);
+        // p2 nothing assigned yet
+        createClient(p2, 1);
+        // p3 nothing assigned but has previous standbyTask
+        final ClientState<TaskId> c3 = createClientWithPreviousActiveTasks(p3, 1);
+        c3.addPreviousStandbyTasks(Utils.mkSet(task00));
+
+        final StickyTaskAssignor<Integer> taskAssignor = createTaskAssignor(task01, task00);
+
+        taskAssignor.assign(0);
+
+        assertThat(clients.get(p3).activeTasks(), equalTo(Collections.singleton(task00)));
+        assertThat(clients.get(p2).activeTasks(), equalTo(Collections.singleton(task01)));
+    }
+
+    @Test
+    public void shouldAssignStandbyTasksToClientThatDontHaveSameActiveTask() throws Exception {
+        final TaskId task03 = new TaskId(0, 3);
+        createClientWithActiveTasks(p1, 1, task00);
```

Comment:
Strictly speaking we don't need 4, but i wanted to mix it up a bit as 3 generally used throughout. Good to try something different.